components/script/dom/webglrenderingcontext.rs

```diff
@@ -437,11 +437,10 @@ impl WebGLRenderingContext {
     fn mark_as_dirty(&self) {
         // If we don't have a bound framebuffer, then don't mark the canvas
         // as dirty.
-        match self.bound_framebuffer.get() {
-            Some(_fb) => self.canvas
-                             .upcast::<Node>()
-                             .dirty(NodeDamage::OtherNodeDamage),
-            None => ()
+        if !self.bound_framebuffer.get().is_none() {
```

Comment:
This is still checking for the opposite condition.

Reply:
Hmm...I got confused between your original comment (and the spec) and your later comment:
"For Servo, that means that the canvas object should **not** be marked dirty if there is no bound framebuffer."
AND
"only mark as dirty when there is no bound framebuffer"
So `is_none()` returns false if the framebuffer is bound, and the `!` negates it to true and marks the canvas object as dirty.

Reply:
It may be better here to use `is_none()` or `is_some()`, instead of inverting the conditional using `!`.